torch/_export/non_strict_utils.py

```diff
@@ -22,19 +23,42 @@
     ShapeEnv,
     StatelessSymbolicContext,
 )
+from torch.utils._pytree import (
+    GetAttrKey,
+    KeyPath,
+    MappingKey,
+    SequenceKey,
+    tree_map_with_path,
+)
 
 
-def fakify(mode, t, t_constraints, source, sources):
+def key_path_to_source(kp: KeyPath) -> Source:
     """
-    Make a fake tensor, given a fake mode, a tensor, constraints indexed
-    by tensor ids, the source for the tensor, and an accumulator mapping
-    tensor dimensions to their sources.
+    Given a key path, return the source for the key path.
     """
+    source: Source = LocalSource("args")
+    for k in kp:
+        if isinstance(k, SequenceKey):
+            source = GetItemSource(source, k.idx)
+        elif isinstance(k, MappingKey):
+            source = GetItemSource(source, k.key)
+        elif isinstance(k, GetAttrKey):
+            source = AttrSource(source, k.name)
```

Comment:
nit: add a catch-all error branch here?

Reply:
@suo doesn't seem like this is addressed lol, but not blocking for me.

Reply:
oh somehow this change got eaten. Let me add it